packages/client/tests/functional/chunking-query/tests.ts

```diff
@@ -0,0 +1,263 @@
+              {
+                id: { in: ids },
+              },
+            ],
+          },
+        })
+      }
+
+      describeIf(providerFlavor === undefined)('With Rust drivers only', () => {
+        // See: https://github.com/prisma/prisma/issues/21802.
+        test('Selecting 32767 ids at once in two inclusive disjunct filters results in error: "too many bind variables", but not with mysql', async () => {
+          const ids = generatedIds(32767)
+
+          try {
+            await selectWith2InFilters(ids)
+
+            if (!['mysql'].includes(provider)) {
+              // unreachable
+              expect(true).toBe(false)
+            }
+          } catch (error) {
+            const e = error as Error
+
+            if (['postgresql', 'cockroachdb'].includes(provider)) {
+              expect(e.message).toContain('Assertion violation on the database')
+              expect(e.message).toContain('too many bind variables in prepared statement')
+              expect(e.message).toContain(`expected maximum of 32767, received 65534`)
+            } else {
+              // unreachable
+              expect(true).toBe(false)
+            }
+          }
+        })
+
+        test('Selecting 32768 ids at once in two inclusive disjunct filters results in error: "too many bind variables"', async () => {
+          const ids = generatedIds(32768)
+
+          try {
+            await selectWith2InFilters(ids)
+            // unreachable
+            expect(true).toBe(false)
+          } catch (error) {
+            const e = error as Error
+
+            if (['postgresql', 'cockroachdb'].includes(provider)) {
+              expect(e.message).toContain('Assertion violation on the database')
+              expect(e.message).toContain('too many bind variables in prepared statement')
+              expect(e.message).toContain(`expected maximum of 32767, received 65535`)
+            } else {
+              expect(e.message).toContain('Prepared statement contains too many placeholders')
+            }
+          }
+        })
+      })
+
+      describeIf(providerFlavor !== undefined)('With Driver Adapters only', () => {
+        test('Selecting 32768 ids at once in two inclusive disjunct filters works', async () => {
+          const ids = generatedIds(32768)
+          await selectWith2InFilters(ids)
+        })
+
+        // See: https://github.com/prisma/prisma/issues/21803.
+        test('Selecting 65536 ids at once in two inclusive disjunct filters results in error', async () => {
+          const ids = generatedIds(65536)
+
+          try {
+            await selectWith2InFilters(ids)
+            // unreachable
+            expect(true).toBe(false)
+          } catch (error) {
+            const e = error as Error
+
+            if (['postgresql', 'cockroachdb'].includes(provider)) {
+              expect(e.message).toContain('bind message has 32767 parameter formats but 0 parameters')
+            }
+          }
+        })
```

Comment:
```suggestion
        test('Selecting 65536 ids at once in two inclusive disjunct filters results in error', async () => {
          expect.assertions(1)
          const ids = generatedIds(65536)

          try {
            // this should throw 
            await selectWith2InFilters(ids)
          } catch (error) {
            const e = error as Error

            if (['postgresql', 'cockroachdb'].includes(provider)) {
              expect(e.message).toContain('bind message has 32767 parameter formats but 0 parameters')
            }
          }
        })
```
When adding expects inside a `try catch` we can add an expectation for the number of assertions `expect.assertions(1)`
Also, it's the pattern we follow in other tests so for clarity and consistency it's better I think.

Reply:
While `expect.assertions(_)` is used elsewhere for the reasons you suggest, in this case, it would add more noise than benefits. Why? Because the `expect()` in the `catch` branch is only called conditionally, so the number of expected assertions should also be computed conditionally. Thanks for the comment, though